File: python/p070.py
import utils


def is_perm(a, b):
    return sorted(str(a)) == sorted(str(b))

best = (10000, 1)
primes = [ i for i in utils.primes(4000) if i > 2000 ]

for i in primes:
    for j in primes:
        n = i * j
        if n > 10**7:
            break
        phi = (i - 1) * (j - 1)
        ratio = (n * 1.0) / phi
        curr = (ratio, n)
        if is_perm(n, phi) and curr < best:
            best = curr

print best[1]

File: erlang/src/p070.erl
-module(p070).

-export([start/0]).

f() ->
  Max = utils:pow(10, 7),
  Primes = [ P || P <- utils:primes(4000), P > 2000 ],
  element(2, lists:min([ {(A * B) / phi(A, B), A * B} || A <- Primes,
                                                         B <- Primes,
                                                         is_perm(A * B, phi(A, B)),
                                                         A * B < Max ])).
is_perm(A, B) ->
  lists:sort(integer_to_list(A)) == lists:sort(integer_to_list(B)).

phi(A, B) ->
  (A - 1) * (B - 1).

start() ->
  io:format("~p ~n", [f()]).
